tests/dbus/_misc.py

```diff
@@ -75,10 +78,127 @@ def __init__(self, not_sim=False):
 
     def setUp(self):
         if MODE == SIMULATOR and not self._not_sim:
-            self._stratisd = subprocess.Popen([os.path.join(_STRATISD), '--sim'])
+            self._stratisd = subprocess.Popen([os.path.join(_STRATISD),
+                                               '--sim'])
         else:
             self._stratisd = subprocess.Popen([os.path.join(_STRATISD),
                                                '--debug'])
 
 
 Service = ServiceR
+
+
+BIN = os.getenv('STRATIS_LOSETUP_BIN', "/usr/sbin/losetup")
+
+
+class LoopBackDevices(object):
+    """
+    Class for creating and managing loop back devices which are needed for
+    specific types of udev event testing.
+    """
+
+    def __init__(self):
+        """
+        Class constructor which creates a temporary directory to store backing
+        file in.
+        """
+        self.dir = tempfile.mkdtemp('_stratis_loop_back')
+        self.count = 0
+        self.devices = {}
+
+    def create_device(self, size_mib):
+        """
+        Create a new loop back device, sparse backing file and attaching it.
+
+        Note: The first time a loop back device is known it will generate
+        a udev "add" event, subsequent backing file changes do not, thus we
+        will need to generate it synthetically.
+        :param size_mib:
+        :return: opaque handle, done as device representing block device will
+                 change.
+        """
+        backing_file = self.dir + '/block_device_%d' % self.count
+        self.count += 1
+
+        with open(backing_file, 'ab') as bd:
+            bd.truncate(size_mib * (1024 * 1024))
```

Comment:
With justbytes:

```size_mib.convertTo(B).numerator``` assuming you were absolutely confident that you got a whole number of bytes, which you  should b eable to control.